File: src/macos/AzPinTests/Utilities/PortalURLTests.swift
import XCTest
@testable import AzPin

final class PortalURLTests: XCTestCase {
    func test_resourceURL_constructedCorrectly() {
        let id = "/subscriptions/sub-1/resourceGroups/rg-1/providers/Microsoft.Web/sites/my-app"
        let url = PortalURL.resource(id: id)
        XCTAssertEqual(url.absoluteString, "https://portal.azure.com/#resource\(id)")
    }

    func test_resourceGroupURL_constructedCorrectly() {
        let url = PortalURL.resourceGroup(subscriptionId: "sub-1", name: "rg-1")
        XCTAssertEqual(url.absoluteString, "https://portal.azure.com/#resource/subscriptions/sub-1/resourceGroups/rg-1")
    }

    func test_resourceId_notDoublePrefixed() {
        let id = "/subscriptions/sub-1/resourceGroups/rg-1/providers/Microsoft.Web/sites/app"
        let url = PortalURL.resource(id: id)
        XCTAssertFalse(url.absoluteString.contains("/#resource/subscriptions/sub-1/resourceGroups/rg-1/providers/Microsoft.Web/sites/app/subscriptions"))
    }
}

File: src/windows/AzPin.Windows.Tests/Tests/PortalUrlTests.cs
using AzPin.Windows.Utilities;

namespace AzPin.Windows.Tests.Tests;

public class PortalUrlTests
{
    [Fact]
    public void ForResource_ContainsResourceId()
    {
        var uri = PortalUrl.ForResource("/subscriptions/s1/resourceGroups/rg1/providers/Microsoft.Web/sites/myapp");
        Assert.Contains("/subscriptions/s1", uri.ToString());
        Assert.Contains("myapp", uri.ToString());
    }

    [Fact]
    public void ForResourceGroup_ContainsSubscriptionAndName()
    {
        var uri = PortalUrl.ForResourceGroup("sub-123", "my-rg");
        var s = uri.ToString();
        Assert.Contains("sub-123", s);
        Assert.Contains("my-rg", s);
        Assert.StartsWith("https://portal.azure.com", s);
    }
}
